experimental/JuLie/tableaux.jl

```diff
@@ -0,0 +1,717 @@
+box, is the number of boxes to the right in the same row + the number
+of boxes below in the same column + 1. The function returns the hook
+length of the box with coordinates `(i,j)`. The functions assumes that
+the box exists.
+"""
+function hook_length(lambda::Partition, i::Integer, j::Integer)
+  h = lambda[i] - j + 1
+  k = i + 1
+  while k<=length(lambda) && lambda[k]>=j
+    k += 1
+    h += 1
+  end
+  return h
+end
+
+@Markdown.doc """
+    hook_length(Tab::Tableau, i::Integer, j::Integer)
+
+Shortcut for ```hook_length(shape(Tab), i, j)```.
+"""
+function hook_length(Tab::Tableau, i::Integer, j::Integer)
+  return hook_length(shape(Tab),i,j)
+end
+
+
+@Markdown.doc """
+    hook_lengths(lambda::Partition)
+
+Returns the tableau of shape ``λ`` in which the entry at position `(i,j)`
+is equal to the hook length of the corresponding box.
+"""
+function hook_lengths(lambda::Partition)
+  if isempty(lambda)
+    return Tableau(Vector{Int}[])
+  end
+  Tab = [ [hook_length(lambda,i,j) for j in 1:lambda[i]] for i in 1:length(lambda) ]
+  return Tableau(Tab)
+end
+
+
+
+
+@doc Markdown.doc"""
+    num_standard_tableaux(lambda::Partition)
+
+Returns the number $f^λ$ of standard tableaux of shape ``λ`` using the hook length formula
+
+$f^λ = \frac{n!}{\prod_{i,j} h_λ(i,j)},$
+
+where the product is taken over all boxes in the Young diagram of ``λ`` and ``h_λ`` denotes the hook length of the box (i,j).
+
+# References
+1. Wikipedia, [Hook length formula](https://en.wikipedia.org/wiki/Hook_length_formula).
+"""
+function num_standard_tableaux(lambda::Partition)
+  n = sum(lambda)
+  h=factorial(ZZ(n))
+  for i = 1:length(lambda)
+    for j = 1:lambda[i]
+      h = div(h,ZZ(hook_length(lambda,i,j)))
+    end
+  end
+  return h
+end
+
+
+@Markdown.doc """
+    schensted(sigma::Vector{Integer})
+    schensted(sigma::Perm{T})
+
+The Robinson–Schensted correspondence is a bijection between
+permutations and pairs of standard Young tableaux of the same shape. For
+a permutation sigma (given as an array), this function performs the
+Schnested algorithm and returns the corresponding pair of standard
+tableaux (the insertion and recording tableaux).
+
+# Example
+```julia-repl
+julia> P,Q = schensted([3,1,6,2,5,4]);
+julia> P
+[[1, 2, 4], [3, 5], [6]]
+
+julia> Q
+[[1, 3, 5], [2, 4], [6]]
+```
+
+# References
+1. Wikipedia, [Robinson–Schensted correspondence](https://en.wikipedia.org/wiki/Robinson–Schensted_correspondence)
+"""
+function schensted(sigma::Vector{T}) where T<:Integer
+  if isempty(sigma)
+    return Tableau(Vector{T}[]),Tableau(Vector{T}[])
+  end
+  P = Tableau{T}([[sigma[1]]])
+  Q = Tableau{T}([[1]])
+  for i = 2:length(sigma)
+    bump!(P, sigma[i], Q, i)
+  end
+  return P,Q
+end
+
+function schensted(sigma::Perm{T}) where T<:Integer
+  return schensted(sigma.d)
+end
+
+
+@Markdown.doc """
+    bump!(Tab::Tableau, x::Int)
+
+Inserts the integer `x` into the tableau Tab according to the bumping
```

Comment:
```suggestion
Inserts the integer `x` into the tableau `Tab` according to the bumping
```
(strictly speaking the argument should be called `tab`, i.e., in all lowercase. Only types use upper case. But again, that's not a hard requirement).

Reply:
I changed the arguments to have lowercase t.